bc_obps/reporting/migrations/0008_schema_data.py

```diff
@@ -0,0 +1,98 @@
+# Generated by Django 5.0.6 on 2024-07-02 21:16
+
+from django.db import migrations
+
+import json
+from registration.models import ReportingActivity
+from reporting.models import SourceType, Configuration
+
+# ACTIVITY
+def init_activity_schema_data(apps, schema_monitor):
+    '''
+    Add initial schema data to erc.activity_schema
+    '''
+    ## Import JSON data
+    import os
+
+    cwd = os.getcwd()
+    with open(f'{cwd}/reporting/json_schemas/2024/general_stationary_combustion/activity.json') as gsc_st1:
+        schema = json.load(gsc_st1)
+
+    ActivitySchema = apps.get_model('reporting', 'ActivityJsonSchema')
+    ActivitySchema.objects.create(
+        reporting_activity_id=ReportingActivity.objects.get(name='General stationary combustion').id,
+        json_schema=schema,
+        valid_from_id=Configuration.objects.get(valid_from='2024-01-01').id,
```

Comment:
@Sepehr-Sobhani Do you have any thoughts on this?

Reply:
I found reasons:
the `import ReportingActivity` here would import the model code from django. There is a chance, because we are running this in a migration, that some changes to that model are not deployed to the database yet and are part of a later migration to be run next.
Using the `apps.get_model("reporting", "ReportingActivity")` guarantees that we get the model object as it's been deployed to the database

Reply:
Exactly ⬆️ . Also, Using `get_model` in Django migrations ensures model decoupling, maintains forward and backward compatibility, and avoids import errors by dynamically loading the model state from the migration creation time.
https://codereview.doctor/features/django/best-practice/avoid-model-imports-in-migrations